20xt/src/base/src/transmit_prioritizer.cpp

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include <mqueue.h>
+#include <unistd.h>
+#include <vector>
+
+#include "mains/transmit_prioritizer.h"
+#include "baja_live_comm.pb.h"
+#include "ipc_config.h"
+#include "live_comm_factory.h"
+#include "xbee/xbee_baja_network_config.h"
+
+static Observation* remainder_data = nullptr;
+
+// Serve as a distributer between the TRXProtoQueues and the POSIX mqueues
+int main()
+{
+  std::cout << "starting transmit prioritizer" << std::endl;
+  
+  // Open queues
+  const mqd_t radio_queue = BajaIPC::open_queue(StationIPC::XBEE_DRIVER_TO_TX_QUEUE, false);
+  if (radio_queue == -1) {
+    std::cout << "Failed to get radio queue. Errno " << errno << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  const std::vector<mqd_t> data_queues = {
+    BajaIPC::open_queue(StationIPC::RTK_CORRECTOR_TX_QUEUE, false),
+    BajaIPC::open_queue(StationIPC::PIT_COMMANDS_TX_QUEUE, false),
+  };
+  for (int i = 0; i < data_queues.size(); i++) {
+    if (data_queues[i] == -1) {
+      std::cout << "Failed to get data queue #" << i << " Errno " << errno << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // We're using POSIX queues, so sending a message is NOT STATELESS
+  while (true) {
+    usleep(100000);
+    std::string msg = build_message(data_queues);
+    if (msg.empty()) {
+      continue;
+    }
+
+    // DEBUG
+    std::cout << "built and enqueuing message of size " << msg.size() << std::endl;
+    LiveComm live_comm;
+    live_comm.ParseFromString(msg);
+    std::cout << live_comm.DebugString() << std::endl;
+    // END DEBUG
+
+    int result = BajaIPC::send_message(radio_queue, msg);
+    if (result == BajaIPC::QUEUE_FULL) {
+      BajaIPC::get_message(radio_queue);
+      result = BajaIPC::send_message(radio_queue, msg);
+    }
+    if (result == BajaIPC::SEND_ERROR) {
+      std::cerr << "Could not enqeue message" << std::endl;
+    }
+  }
+
+  return EXIT_SUCCESS;
+}
+
+/* Scan through outgoing-data queues and build a payload. Return empty str if not data found */
+std::string build_message(const std::vector<mqd_t>& data_queues)
+{
+  LiveCommFactory factory = LiveCommFactory();
```

Comment:
Continued from other comments. I would like to remove the current packing method in LiveCommFactory().

It doesn't make much sense to aggregate multiple Observations into a single Observation within the context of this file. Unnecessary complexity, marginal performance gains, and likely to break.